python/amici/ode_export.py

```diff
@@ -2027,9 +2027,10 @@ def _derivative(self, eq: str, var: str, name: str = None) -> None:
             nonzeros = np.asarray(
                 derivative.applyfunc(lambda x: int(not x.is_zero))
             ).astype(np.int64)
-            if max(nonzeros.shape):
-                while nonzeros.max():
-                    nonzeros = nonzeros.dot(nonzeros)
+            recursion = nonzeros.copy()
+            if max(recursion.shape):
+                while recursion.max():
```

Comment:
```suggestion
                while recursion.max():
```
Does this need `abs`?

Reply:
no. `recursion` in effectively counts the number of paths of length `self._w_recursion_depth` from i to j, so the number is always positive.